### PR TITLE
Escape % in message and values

### PR DESCRIPTION
**Pull request by msabramo**
Before:

```
... URL='http://192.168.223.128:8083/api/v1/networkpolicies/mattapp%!F(MISSING)mattimage'
```

After:

```
... URL='http://192.168.223.128:8083/api/v1/networkpolicies/mattapp%2Fmattimage'
```

Fixes: https://github.com/adobe-community/issues-ethos/issues/4284

Cc: @iderdik, @jimmyislive

### CODE REVIEW

File: formatters/kv.go [hunk 1]
```diff
@@ -19,9 +19,18 @@ func KVEntryString(entry *logrus.Entry) string {
 		keys = append(keys, k)
 	}
 
+	// Escape % if it appears in value, so Sprintf doesn't try to expand it
+	entry.Message = strings.Replace(entry.Message, "%", "%%", -1)
```
Comment (iderdik) [inline formatters/kv.go]: what if  there are already 2? This seems like an inherent flaw in sprintf

Comment (msabramo) [reply formatters/kv.go]: I guess we could add a line before this that converts `"%%"` to `"{double_percent}"` 😄  ?

Comment (iderdik) [reply formatters/kv.go]: Or we could just use string concatenation. Is there a downside to that?

**Comment by iderdik** [comment]
👍

File: formatters/kv.go (hunk 1 above)
Comment (msabramo) [reply formatters/kv.go]: Oh that's a good idea.

**Comment by msabramo** [comment]
@iderdik: 5df5c4e changes to use string concatenation. And I added tests! 😄 

I know you already thumbed it, but since I changed it quite a bit, want to give it another quick once over and make sure I didn't miss anything?

**Review by iderdik — APPROVED** [review]

**Comment by msabramo** [comment]
Hmmm I don't seem to have the ability to merge.

**Comment by msabramo** [comment]
Thanks, @iderdik!

**Pull request merged**